app/api/metrics.py

```diff
@@ -51,3 +63,61 @@ def getScoreCards(
     teamId = decodeJWT(token)["teamId"]
 
     return microserviceScoreCardService.getByTeamId(teamId)
+"""
+
+# ADD NEW METRIC HERE
+@router.post("/", response_model=schemas.Metric)
+def createMetric(metric: schemas.MetricCreate, metricCrud: crud.CRUDMetric = Depends(dependencies.getMetricsCrud)) -> schemas.Metric:
+    metricObj = metric
+    # change ' ' with '-'
+    metricObj.code = format_code(metric.name)
+    # Stringfying the list of areas to be saved as string 
+    metricObj.area = json.dumps(metric.area)
+    # Handling the datatype of the field type to be integer or boolean 
+    if (metricObj.type != "integer" and metricObj.type != "boolean"):
```

Comment:
this will not extend smoothly when you want to add a new metric type.
suggestion: define a list of acceptable types, and update the if condition to check if the metricObj.type is within the defined list or not